packages/components/src/internal/components/editable/EditableGrid.tsx

```diff
@@ -89,6 +89,70 @@ function moveUp(colIdx: number, rowIdx: number): CellCoordinates {
     return { colIdx, rowIdx: rowIdx - 1 };
 }
 
+/**
+ * Computes the new range for a given grid dimension when expanding or contracting in a particular direction.
+ * @param selectedIdx: The index of the currently selected cell
+ * @param min: The minimum of the current range
+ * @param max: The maximum of the current range
+ * @param direction: number in the range of -1, 1.
+ *  - If -1, we are moving up or left
+ *  - If 0 we are not moving
+ *  - If 1 we are moving down or right
+ */
+function computeRangeChange(selectedIdx: number, min: number, max: number, direction: number): [number, number] {
```

Comment:
Clean. Seems like it is worth considering a unit test to cover edge cases.

Reply:
Done